pkg/policy/l4.go

```
@@ -220,41 +225,47 @@ func (l7 L7DataMap) addRulesForEndpoints(rules api.L7Rules, endpoints []api.Endp
 			l7[epsel] = rules
 		}
 	} else {
-		// If there are no explicit fromEps, have a 'special' wildcard endpoint.
-		l7[api.WildcardEndpointSelector] = rules
+		panic("Must have at least the wildcard selector")
```

Oof, panic? Why not return an error and fail accordingly in callers?

---

If it's practical to throw all paths of `L7DataMap` construction in some tests then I'm not super worried about this, but I agree that seeing the `panic` here makes not only the code panic, but also the developers ;)